tests/csapi/apidoc_room_event_test.go

```diff
@@ -0,0 +1,159 @@
+package csapi_tests
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+)
+
+func stringSliceContains(slice []string, s string) bool {
+	for _, value := range slice {
+		if value == s {
+			return true
+		}
+	}
+	return false
+}
+
+// sytest: Events come down the correct room
+func TestEventsInCorrectRoom(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	const roomAmount = 30
+
+	// Create all rooms
+
+	var chanRooms = make(chan string, roomAmount)
+
+	// semaphore, to reduce load on the server when creating rooms
+	// on slow PCs, increasing this will time out the test :(
+	var sem = make(chan bool, 1)
+
+	for i := 0; i < roomAmount; i++ {
+		sem <- true
+
+		go func() {
+			roomId := alice.CreateRoom(t, map[string]interface{}{
+				"preset": "public_chat",
+			})
+
+			chanRooms <- roomId
+
+			<-sem
+		}()
+	}
+
+	var rooms []string
+
+	for i := 0; i < roomAmount; i++ {
+		// This implicitly waits until all rooms are done creating
+		rooms = append(rooms, <-chanRooms)
```

Comment:
You can/should be `range`ing over `chanRooms` and close it once the last room has been sent, rather than do this.

Reply:
That's hard to do with a large amount of concurring goroutines, so I'm not sure how to do this.